src/client/csfle/state_machine.rs

```diff
@@ -40,9 +40,9 @@ impl CryptExecutor {
         key_vault_namespace: Namespace,
         tls_options: Option<KmsProvidersTlsOptions>,
     ) -> Result<Self> {
-        let num_cpus = std::thread::available_parallelism()?.get();
+        // TODO(RUST-1492): Replace num_cpus with std::thread::available_parallelism.
         let crypto_threads = rayon::ThreadPoolBuilder::new()
-            .num_threads(num_cpus)
+            .num_threads(num_cpus::get())
```

Comment:
On Linux, this appears to [do file I/O](https://github.com/seanmonstar/num_cpus/blob/master/src/linux.rs#L60) to read `/proc/cpuinfo`. I doubt this is a huge problem, but just to confirm: this isn't generally called in a hot path or per operation, right? Typically this is done once per client?

Reply:
Correct, once per `Client` that has encryption enabled (not run at all for clients that don't), and once per `ClientEncryption`.  Neither of those are in a hot path.